From 85bc3deb0118207a8eb470a56139ce0581782e3b Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Wed, 09 Jun 2021 14:17:31 +0300
Subject: [PATCH] Revert "Revert "Revert "[SW-41316] preboot: make flexible fw load setting platform specific"""

This reverts commit 07c842e83fd7e8d2e392ee7421f4abd037554c64.

Dynamic FW loading was showing issue on Greco due to problem in scrambling
enablement sequence by LKD. The issue is now fixed in LKD
(https://gerrit.habana-labs.com/#/c/98689/). So renabling dynamic FW
loading bit.

Change-Id: Ie4b659692dbe2e7b2e3b252dfd22e642056adac7
---

diff --git a/arch/gaudi/common.c b/arch/gaudi/common.c
index f1e3e7c..745079f 100644
--- a/arch/gaudi/common.c
+++ b/arch/gaudi/common.c
@@ -444,7 +444,6 @@
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_HARD_RST_EN, true);
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 /* Hard coded each of the gaudi on-chip TS device properties */
diff --git a/arch/goya/common.c b/arch/goya/common.c
index 9e7d889..6488f2d 100644
--- a/arch/goya/common.c
+++ b/arch/goya/common.c
@@ -25,7 +25,6 @@
 #include <preboot/fit.h>
 #include <preboot/thermal.h>
 #include <string.h>
-#include <boot_if.h>
 
 #include "common.h"
 
@@ -217,11 +216,6 @@
 u32 hl_pll_timestamp_base_get(void)
 {
 	return mmCPU_TIMESTAMP_CNTCR;
-}
-
-void lkd_dev_specific_setup(void)
-{
-	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 }
 
 #define GOYA_ASIC_TS_MAX_COUNT 3
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index c67e44e..8e2524c 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -31,6 +31,7 @@
 
 void module_lkd_enable(void)
 {
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_FW_LD_COM_EN, true);
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
